### PR TITLE
Add type dependencies for mypy

### PR DESCRIPTION
**Pull request by pieterhelsen**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Before submitting a PR make sure that your feature is covered with tests.
 
 ```
 pip3 install black flake8 mypy
+pip3 install types-PyYAML types-python-dateutil types-paramiko types-retry
```
Comment (martomi) [inline CONTRIBUTING.md:38]: Do you know why it suddenly became necessary to install these separately?

**Review by martomi — APPROVED** [review]
Thanks!

File: CONTRIBUTING.md (hunk 1 above)
Comment (pieterhelsen) [reply CONTRIBUTING.md:38]: Think it's related to this blogpost:
https://mypy-lang.blogspot.com/2021/05/the-upcoming-switch-to-modular-typeshed.html

Comment (martomi) [reply CONTRIBUTING.md:38]: Seems there's also the command `mypy --install-types` which automatically detects and installs the missing types. Something for the future perhaps.

**Pull request merged**